### PR TITLE
fix: emptyMessage prop and default label in unstyled mode

### LINKED ISSUE #6671: Dropdown: emptyMessage prop doesn't work
### Describe the bug

Using a Dropdown with unstyled mode shows a 'empty' label when no option is selected. Prop 'emptyMessage' to customize that message doesn't work.

```tsx
<Dropdown
  id="gender"
  name="gender"
  value={gender}
  onChange={e => setGender(e.target.value)}
  options={genders}
  emptyMessage="Selecciona una opción"
  unstyled
/>
```

error_1
error_2

Also, as I'm using the dropdown with a float label, both 'empty' label and my defined float label they overlap each other.

```tsx
<span className="p-float-label">
  <Dropdown
    id="gender"
    name="gender"
    value={gender}
    onChange={e => setGender(e.target.value)}
    options={genders}
    emptyMessage="Selecciona una opción"
    unstyled
  />
  <label htmlFor="gender">Género *</label>
</span>
```

error_3

### Reproducer

_No response_

### PrimeReact version

10.6.6

### React version

18.x

### Language

TypeScript

### Build / Runtime

Next.js

### Browser(s)

_No response_

### Steps to reproduce the behavior

_No response_

### Expected behavior

If I don't use emptyMessage prop it shouldn't show any label.
And if I define emptyMessage prop it should show that label.

### PR DESCRIPTION
**Pull request by ivanpajon**
### Defect Fixes
Fix #6671 

When no emptyMessage is defined and no option is selected the component internally use a `&nbsp;` so it not loses its height.

Without `&nbsp;`:
without nbsp

With `&nbsp;`:
with nbsp

### CODE REVIEW

File: components/lib/dropdown/Dropdown.js [hunk 1]
```diff
@@ -1109,7 +1109,7 @@ export const Dropdown = React.memo(
                 return <input {...inputProps} />;
             }
 
-            const content = props.valueTemplate ? ObjectUtils.getJSXElement(props.valueTemplate, selectedOption, props) : label || props.placeholder || 'empty';
+            const content = props.valueTemplate ? ObjectUtils.getJSXElement(props.valueTemplate, selectedOption, props) : label || props.placeholder || props.emptyMessage || '';
```
Comment (melloware) [inline components/lib/dropdown/Dropdown.js:1112]:
```suggestion
            const content = props.valueTemplate ? ObjectUtils.getJSXElement(props.valueTemplate, selectedOption, props) : label || props.placeholder || props.emptyMessage || <>&nbsp;</>;
```

File: components/lib/dropdown/Dropdown.js [hunk 2]
```diff
@@ -1119,7 +1119,7 @@ export const Dropdown = React.memo(
                 ptm('input')
             );
 
-            return <span {...inputProps}>{content}</span>;
+            return <span {...inputProps}>{content || <>&nbsp;</>}</span>;
```
Comment (melloware) [inline components/lib/dropdown/Dropdown.js:1122]:
```suggestion
            return <span {...inputProps}>{content}</span>;
```

**Review by melloware — CHANGES_REQUESTED** [review]
I made a couple of suggestions. WDYT?

**Comment by ivanpajon** [comment]
It looks cleaner, I like it. Thanks!

**Review by melloware — APPROVED** [review]

**Pull request merged**